### PR TITLE
BXMSDOC-3368 - Validate, correct, and apply new KIE Server REST API e…

### PR DESCRIPTION
**Pull request by mswiderski**
…ndpoint data to Swagger

@sterobin @MarianMacik could you please have a look? This is the bulk of the updates of REST api of Kie Server.

### CODE REVIEW

**Comment by sterobin** [comment]
@mswiderski, I replied in the [JIRA](https://issues.jboss.org/browse/BXMSDOC-3368). A lot done on this! Some remaining things left.

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
It seems that the test failures are being fixed by #1721.
So we should wait and retest after it is merged.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/build/revapi-config.json [hunk 1]
```diff
@@ -101,7 +101,20 @@
            "methodName": "updateReleaseId",
            "elementKind": "method",
            "justification": "AF-1685: Development Lifecycle Streamline"
+         },
+		 {
```
Comment (yurloc) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/build/revapi-config.json]: Tabs.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-drools/src/main/java/org/kie/server/remote/rest/drools/CommandResource.java [hunk 2]
```diff
@@ -15,6 +15,12 @@
 
 package org.kie.server.remote.rest.drools;
 
+import static org.kie.server.remote.rest.common.util.RestUtils.buildConversationIdHeader;
+import static org.kie.server.remote.rest.common.util.RestUtils.createResponse;
+import static org.kie.server.remote.rest.common.util.RestUtils.getClassType;
+import static org.kie.server.remote.rest.common.util.RestUtils.getContentType;
+import static org.kie.server.remote.rest.common.util.RestUtils.getVariant;
```
Comment (yurloc) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-drools/src/main/java/org/kie/server/remote/rest/drools/CommandResource.java]: Static imports should be at the bottom.

Comment (yurloc) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-drools/src/main/java/org/kie/server/remote/rest/drools/CommandResource.java]: OK, I just realized the rest of the code has static imports first, so let's keep it this way for the sake of consistency.

We should either apply KIE Java Conventions everywhere or accept that different repos/projects follow different conventions, in which case those conventions should be declared, ready to be imported to mainstream IDEs and automatically **checked**.

**Comment by MarianMacik** [comment]
I have resolved the conflicts and rebased it, so now the tests will be rerun.

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by yurloc** [comment]
Please re-check the failing tests. At least this seems to be related to the changes in this PR:
```
expected:<Missing value for p[]InstanceId> but was:<Missing value for p[rocess]InstanceId>
```

**Comment by MarianMacik** [comment]
@yurloc Yes, there are some issues mainly regarding instance migration, I have it on my list.

**Pull request merged**